### ISSUE TITLE
Couldn't read data for number error constantly

### ISSUE THREAD

**Issue by mdegat01** [bbernhard/signal-cli-rest-api#243]
### The problem

I've been having some issues with this container. I use it from Home Assistant by way of the addon primarily as a way of notifying other folks in my home who prefer texts over mobile notifications. Since I'm not usually the recipient of these texts sometimes it takes me a while to notice (i.e. one day they will say "oh I haven't received texts for like 2 weeks" 😒 ). I noticed streams of messages like this in the log:
```
time="2022-05-07T10:49:48-04:00" level=error msg="Couldn't read data for number +12345678900: EOF. Is the number properly registered?
```
Restarting the container fixes it immediately. The number is in fact properly registered, that's not the issue. As soon as I restart the container I can send test texts fine and everything works again, no errors.

I have all my logs for the past month accessible so I've been going through it and I noticed a pattern. The start of this error seems to line up with a host restart. What I believe happens is that Node RED opens a websocket connection to the container as soon as it starts up. I think it does this before the container is finished whatever it has to do to get ready. When this happens it seems to break the container (or just the number? I only have one so I don't know) in some way that can only be fixed by a restart. That also explains why a restart fixes it because Node RED seems to have a 5 minute retry after a failure so then it gives it time to start up properly.

I'm going to add in a workaround to test my theory. But I wanted to report in my findings as this has been plaguing me for a bit and might be creating issues for others too.

### Are you using the latest released version?

- [X] Yes

### Have you read the troubleshooting page?

- [X] Yes

### What type of installation are you running?

Home Assistant Signal Addon

### In which mode are you using the docker container?

JSON-RPC Mode

### What's the architecture of your host system?

arm64

**Comment by bbernhard** [comment]
Thanks a lot for the detailed description - very much appreciated!

In case your theory confirms, I would be really grateful if you could let me know. That definitely would help to track the issue down.

**Comment by mdegat01** [comment]
So I can confirm that it is definitely related to startup. I set up a monitor to watch for that line in the signal addons logs and restart it and notify me when it sees it. Every time I restart the host I see that error immediately following startup and it goes away once the addon is restarted.

However it isn't related to Node RED. I turned off Node RED and restarted the host and that made no difference. And as far as I know nothing else is connecting to the container. HA doesn't open a websockets connection, all it does is call the `v2/send` API when it wants to send something (which is far after restart).

So I guess now I'm wondering if its addon specific. In fact in looking closer it might be [this line](https://github.com/haberda/signal-addon/blob/main/signal/Dockerfile#L19). HA's supervisor already makes `/data` a volume in all addons so I'm not sure what declaring it again does.

I'll leave this open for now until I'm sure what's going on, will close if it turns out to be purely the addon. Although if you have any other ideas for how I might debug this let me know.

**Comment by bbernhard** [comment]
Thanks a lot for the detailed description!

This indeed sounds really strange. If I understood you correctly, the only way to provoke the issue is to reboot the host system? If you manually restart the HA Signal addon the error is gone? 

Unfortunately, I do not know much about Home Assistant's addon management, so I am not sure if this is actually possible, but can you somehow connect into the running docker container? 

Normally I would do it like this:

* get container id with `docker ps`
* run `docker run -it <container id> /bin/bash`

But not sure if that's also possible with the HA addons?

I would be really curious to see what's going on inside the container when the error occurs.

If you find a way to connect to the running docker container I would be interested in:

* the log files inside the `/var/log/signal-cli-json-rpc-0` folder (the log might contain some sensitive information, please make sure to obfuscate any phone numbers and other personal information)
* if you check with `ps`, do you see a signal process running? 

Just out of curiosity: Have you tried temporarily switching to the normal/native mode? Does the issue also happen there?

At the moment I have two signal-cli-rest-api instances running on two different systems and I haven't experienced any issues after a reboot so far. So either your setup somehow triggers a bug in the REST API wrapper or - as you already mentioned - it could also be addon specific. :thinking:

**Comment by aneisch** [comment]
Looks like this has gone stale, but this same "could not read data.." log has started for me today and no amount of container stopping seems to resolve it. 

```
root@c7b101da82d4:/# ps 
  PID TTY          TIME CMD
  181 pts/0    00:00:00 bash
  225 pts/0    00:00:00 ps
```

```
root@c7b101da82d4:/# tail -1 /var/log/signal-cli-json-rpc-1/out.log
Error while checking account +$MY_NUMBER: StatusCode: 499
```

This seems to maybe indicate HTTP Status code 499 ("client closed request") but maybe that's a bad assumption on my part. 

Another interesting anecdote from when I noticed the error due to Home Assistant not being able to send a message through the API before I started messing with the signal-cli container:
```
2022-10-05 15:32:27.084 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Unexpected error for call_service at pos 1: Failed to send message: java.util.concurrent.ExecutionException: org.whispersystems.signalservice.api.push.exceptions.DeprecatedVersionException: StatusCode: 499 (IOException)
```

Not sure if "org.whispersystems.signalservice.api.push.exceptions.DeprecatedVersionException" is something related to a deprecated signal client version in the container (I am using :latest) or maybe a false flag and another bad assumption on my part.

**Comment by etlweather** [comment]
For a quick and dirty fix, see https://github.com/AsamK/signal-cli/issues/1022#issuecomment-1269169653

**Comment by rfrederick** [comment]
I've had the same issue since around the same timestamp. Currently the only piece of routine functionality that's working for me is sending to groups.

**Comment by aneisch** [comment]
> For a quick and dirty fix, see [AsamK/signal-cli#1022 (comment)](https://github.com/AsamK/signal-cli/issues/1022#issuecomment-1269169653)

Temporary fix for this repo specifically: https://github.com/bbernhard/signal-cli-rest-api/issues/286#issuecomment-1269290569
